### ISSUE TITLE
temperatureDependent

### ISSUE THREAD

**Issue by senlinq** [DLR-RY/TwoPhaseFlow#5]
If sigma is temperatureDependent, then there may a tangential gradient of sigma. Perhaps _surfaceTensionForce()_  in **surfaceTensionForceModel.H** is lack of the tangential gradient term?
<img width="318" alt="tangential" src="https://user-images.githubusercontent.com/67735937/123509075-0e6ef500-d6a6-11eb-83dc-0fbb5b2f7ceb.PNG">

**Comment by HenningScheufler** [comment]
This is correct. I added the temperature depended surface tension for completeness and neglected this term in the current implementation. This should be pointed out in the documentation.


references:
https://www.researchgate.net/profile/Roland-Engberg-2/publication/258698745_The_impact_of_Marangoni_convection_on_fluid_dynamics_and_mass_transfer_at_deformable_single_rising_droplets_-_A_numerical_study/links/5cff6f1a4585157d15a214e6/The-impact-of-Marangoni-convection-on-fluid-dynamics-and-mass-transfer-at-deformable-single-rising-droplets-A-numerical-study.pdf

https://reader.elsevier.com/reader/sd/pii/002199919290240Y?token=2AB60B0F79D64FB7E6C8C6571DE393E2B081C7BD9E937BB833E52088EBB136E43EBAA4B1AF13A7751382B96E54AF4ABB&originRegion=eu-west-1&originCreation=20210628125640

**Comment by senlinq** [comment]
> This is correct. I added the temperature depended surface tension for completeness and neglected this term in the current implementation. This should be pointed out in the documentation.
> 
> references:
> https://www.researchgate.net/profile/Roland-Engberg-2/publication/258698745_The_impact_of_Marangoni_convection_on_fluid_dynamics_and_mass_transfer_at_deformable_single_rising_droplets_-_A_numerical_study/links/5cff6f1a4585157d15a214e6/The-impact-of-Marangoni-convection-on-fluid-dynamics-and-mass-transfer-at-deformable-single-rising-droplets-A-numerical-study.pdf
> 
> https://reader.elsevier.com/reader/sd/pii/002199919290240Y?token=2AB60B0F79D64FB7E6C8C6571DE393E2B081C7BD9E937BB833E52088EBB136E43EBAA4B1AF13A7751382B96E54AF4ABB&originRegion=eu-west-1&originCreation=20210628125640

**Comment by senlinq** [comment]
> This is correct. I added the temperature depended surface tension for completeness and neglected this term in the current implementation. This should be pointed out in the documentation.
> 
> references:
> https://www.researchgate.net/profile/Roland-Engberg-2/publication/258698745_The_impact_of_Marangoni_convection_on_fluid_dynamics_and_mass_transfer_at_deformable_single_rising_droplets_-_A_numerical_study/links/5cff6f1a4585157d15a214e6/The-impact-of-Marangoni-convection-on-fluid-dynamics-and-mass-transfer-at-deformable-single-rising-droplets-A-numerical-study.pdf
> 
> https://reader.elsevier.com/reader/sd/pii/002199919290240Y?token=2AB60B0F79D64FB7E6C8C6571DE393E2B081C7BD9E937BB833E52088EBB136E43EBAA4B1AF13A7751382B96E54AF4ABB&originRegion=eu-west-1&originCreation=20210628125640

Thanks a lot for reply. So, is the naive implementation I give in the picture right for tangential gradient term?
<img width="926" alt="tangential" src="https://user-images.githubusercontent.com/67735937/123766773-8d0fa080-d8f9-11eb-8aca-ceab34add678.PNG">
Another question, may the entry of dictionary for sigma be assinged with expressions, such as 
{
type     temperatureDependent
sigma   T-T^2
}
<img width="605" alt="expression" src="https://user-images.githubusercontent.com/67735937/123768482-f2b05c80-d8fa-11eb-8ded-3cdcf062d650.PNG">

**Comment by HenningScheufler** [comment]
```
surfaceForces
{
  sigma
  {
        type    test;
        sigma   0.07;
  }
  surfaceTensionForceModel gradAlpha;
  accelerationModel gravity;
  deltaFunctionModel alphaCSF;
}

outputs:

--> FOAM FATAL IO ERROR: (openfoam-2106)
Unknown surfaceTensionModel type test

Valid surfaceTensionModel types :
3(constant liquidProperties temperatureDependent)
```

**Comment by brucethemuce** [comment]
Makes sense, but how do you prescribe the function of temperature for sigma?

```
  sigma
  {
        type    temperatureDependent;
        sigma   0.07; %how does this translate to a function of T?
  }
```
There is only one coefficient, and two are needed for mX+B form

**Comment by HenningScheufler** [comment]
sigma is a Function1:

https://www.openfoam.com/documentation/guides/latest/api/temperatureDependentSurfaceTension_8H_source.html

sigma is a function of the temperature. You can specify it as polynomial:

```
sigma{ 
    type temperatureDependent;
        sigma polynomial
        (
            (0.001 1)
            (-4e-4 2)
        );
    };
```

```
sigma{ 
    type temperatureDependent;
        sigma test; // crashes a gives options
    };
```


details:
https://www.openfoam.com/documentation/guides/latest/api/classFoam_1_1Function1.html

https://www.openfoam.com/documentation/guides/latest/api/classFoam_1_1surfaceTensionModels_1_1temperatureDependent.html#ab11d65be145df47b2d4871fb31778c28

**Comment by brucethemuce** [comment]
It doesnt seem to work with multiRegionPhaseChangeFlow, the solver crashes immediately. Below is the thermophysicalProperties file for testsuite/cht/fixedFlux. 

```
// * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * //

phases (water air);

pMin            pMin [ 1 -1 -2 0 0 0 0 ] 10000;

sigma           sigma [ 1 0 -2 0 0 0 0 ] 0;

TMin           TMin [ 0 0 0 1 0 0 0 ] 50;
TMax           TMax [ 0 0 0 1 0 0 0 ] 500;

water
{
    type pureMovingPhaseModel;
}

air
{
    type pureMovingPhaseModel;
}

surfaceForces
{
sigma
{
    type        temperatureDependent;
    sigma polynomial
    (
        (0.001 1)
        (-4e-4 2)
    );
}
  surfaceTensionForceModel gradAlpha;
  accelerationModel gravity;
  deltaFunctionModel alphaCSF;
}

// ************************************************************************* //


```
The sample case depthCharge2D for compressibleInterFoam works fine with this
```
phases          (water air);

pMin            10000;

sigma
{
    type        temperatureDependent;
    sigma polynomial
    (
        (0.001 1)
        (-4e-4 2)
    );
}

```

**Comment by HenningScheufler** [comment]
The temperature in testsuite/cht/fixedFlux is 373 K so we get a sigma of:

`sigma = 0.001*T - 4e-4*T**2 = -55`

This should cause a crash in the maxCapillaryNum (negative value under the root). However, this is not implemented in compressibleInterFoam:

You can write out fields, here sigma, from the registry with volFieldValue with:

writeFields     true;

Now, you can check with paraview if the values are correct

https://www.openfoam.com/documentation/guides/latest/doc/guide-fos-field-volFieldValue.html
https://develop.openfoam.com/Development/openfoam/-/blob/master/tutorials/multiphase/icoReactingMultiphaseInterFoam/poolEvaporation/system/controlDict
